src/sig/sig.h

```diff
@@ -63,108 +84,123 @@ struct OQS_SIG {
+	 * compile-time macros `OQS_SIG_*_length_*`.
+	 *
+	 * @param[out] signature The signature on the message represented as a byte string.
+	 * @param[out] signature_len The length of the signature.
+	 * @param[in] message The message to sign represented as a byte string.
+	 * @param[in] message_len The length of the message to sign.
+	 * @param[in] secret_key The secret key represented as a byte string.
+	 * @return OQS_SUCCESS or OQS_ERROR
 	 */
-	OQS_STATUS(*sign)(const OQS_SIG *s, const uint8_t *priv, const uint8_t *msg, const size_t msg_len, uint8_t *sig, size_t *sig_len);
+	OQS_STATUS (*sign)(uint8_t *signature, size_t *signature_len, const uint8_t *message, size_t message_len, const uint8_t *secret_key);
 
 	/**
-	 * Pointer to a function for signature verification.
+	 * Signature verification algorithm.
+	 *
+	 * Caller is responsible for allocating sufficient memory for `message`, based
+	 * on the `length_*` members in this object or the per-scheme compile-time macros
+	 * `OQS_SIG_*_length_*`.
 	 *
-	 * @param s                The signature structure.
-	 * @param pub              The signer's public key.
-	 * @param msg              The signed message.
-	 * @param msg_len          Length of the signed message.
-	 * @param sig              The signature to verify.
-	 * @param sig_len          Length of the signature to verify.
-	 * @return                 OQS_SUCCESS on success, or OQS_ERROR on failure.
+	 * @param[in] message The message represented as a byte string.
+	 * @param[in] message_len The length of the message.
+	 * @param[in] signature The signature on the message represented as a byte string.
+	 * @param[in] signature_len The length of the signature.
+	 * @param[in] public_key The public key represented as a byte string.
+	 * @return OQS_SUCCESS or OQS_ERROR
 	 */
-	OQS_STATUS(*verify)(const OQS_SIG *s, const uint8_t *pub, const uint8_t *msg, const size_t msg_len, const uint8_t *sig, const size_t sig_len);
+	OQS_STATUS (*verify)(uint8_t *message, size_t message_len, const uint8_t *signature, size_t signature_len, const uint8_t *public_key);
 	// clang-format on
 
-	/**
-	 * Pointer to a function for freeing the allocated signature structure
-	 *
-	 * @param s                Signature structure
-	 */
-	void (*free)(OQS_SIG *s);
-};
+} OQS_SIG;
 
 /**
- * Instantiate a new signature object.
+ * Consturcts an OQS_SIG object for a particular algorithm.
+ *
+ * Callers should always check whether the return value is `NULL`, which indicates either than an
+ * invalid algorithm name was provided, or that the requested algorithm was disabled at compile-time.
  *
- * @param algid              The id of the signature algorithm to be instantiated.
- * @return                   A new signature object on success, or NULL on failure.
+ * @param[in] method_name Name of the desired algorithm; one of the names in `OQS_SIG_algs`.
+ * @return An OQS_SIG for the particular algorithm, or `NULL` if the algorithm has been disabled at compile-time.
```

Comment:
... if the algorithm has been disabled at compile-time or is not recognized.

Reply:
We should update kem.h as well...

Reply:
Will do.